src/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -156,41 +156,124 @@ public Task<Stream> GetStreamAsync(string requestUri)
         // Unbuffered by default
         public Task<Stream> GetStreamAsync(Uri requestUri)
         {
-            return GetContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, Stream.Null,
+            return GetStreamContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead,
                 content => content.ReadAsStreamAsync());
         }
 
-        private Task<T> GetContentAsync<T>(Uri requestUri, HttpCompletionOption completionOption, T defaultValue,
-            Func<HttpContent, Task<T>> readAs)
+        private Task<string> GetStringContentAsync(Uri requestUri, HttpCompletionOption completionOption, 
+            Func<HttpContent, Task<string>> readAs)
         {
-            TaskCompletionSource<T> tcs = new TaskCompletionSource<T>();
+            var tcs = new TaskCompletionSource<string>(readAs);
 
-            GetAsync(requestUri, completionOption).ContinueWithStandard(requestTask =>
+            GetAsync(requestUri, completionOption).ContinueWithStandard(tcs, (requestTask, state) =>
             {
-                if (HandleRequestFaultsAndCancelation(requestTask, tcs))
+                var innerTcs = (TaskCompletionSource<string>)state;
+                var innerReadAs = (Func<HttpContent, Task<string>>)innerTcs.Task.AsyncState;
+                if (HandleRequestFaultsAndCancelation(requestTask, innerTcs))
                 {
                     return;
                 }
                 HttpResponseMessage response = requestTask.Result;
                 if (response.Content == null)
                 {
-                    tcs.TrySetResult(defaultValue);
+                    innerTcs.TrySetResult(string.Empty);
```

Comment:
With Roslyn, yes, it should be able to.  Pre-Roslyn, that can be handled by changing the method from being a generic method to being a non-generic method in a generic class.